### ISSUE TITLE
Select only 2D or only 3D models when 'trying all available models'

### ISSUE THREAD

**Issue by wolny** [hci-unihd/plant-seg#142]
When trying all available models, we should only try 2D or 3D models on a given data. We can select the models automatically based on the `patch_size` or input size. Currently if the patch size is 3D the 2d models just fail and the other way around.

**Comment by lorenzocerrone** [comment]
I think it make sense to only try 2D models on 2D inputs, but sometimes it might be useful to run the 2D models on 3D inputs (like if you have very coarse resolution in z, or not enough z-slices, or if you have a 2D+t stack)

**Comment by wolny** [comment]
yeah, I agree to that. Then we should add some simple logic to handle that
